service/integration/registered_resources_test.go

```diff
@@ -587,66 +625,185 @@ func (s *RegisteredResourcesSuite) Test_CreateRegisteredResourceValue_WithNonUni
+						AttributeValueId: invalidID,
+					},
+				},
+			},
+			err: db.ErrForeignKeyViolation,
+		},
+		{
+			name: "Invalid Attribute Value FQN",
+			actionAttrValues: []*registeredresources.ActionAttributeValue{
+				{
+					ActionIdentifier: &registeredresources.ActionAttributeValue_ActionName{
+						ActionName: actions.ActionNameCreate,
+					},
+					AttributeValueIdentifier: &registeredresources.ActionAttributeValue_AttributeValueFqn{
+						AttributeValueFqn: "https://example.com/attr/does_not_exist/value/invalid",
+					},
+				},
+			},
+			err: db.ErrNotFound,
+		},
+	}
+
+	for i, tc := range testCases {
+		s.Run(tc.name, func() {
+			req := &registeredresources.CreateRegisteredResourceValueRequest{
+				ResourceId:            res.GetId(),
+				Value:                 fmt.Sprintf("test_create_res_value_invalid_action_attr_values_%d", i),
+				ActionAttributeValues: tc.actionAttrValues,
+			}
+
+			created, err := s.db.PolicyClient.CreateRegisteredResourceValue(s.ctx, req)
+			s.Require().Error(err)
+			s.Require().ErrorIs(err, tc.err)
+			s.Nil(created)
+		})
+	}
 }
 
-func (s *RegisteredResourcesSuite) Test_GetRegisteredResourceValue_ByFQN_Succeeds() {
+// Get
+
+func (s *RegisteredResourcesSuite) Test_GetRegisteredResourceValue_Valid_Succeeds() {
 	existingRes := s.f.GetRegisteredResourceKey("res_with_values")
 	existingResValue1 := s.f.GetRegisteredResourceValueKey("res_with_values__value1")
-	fqn := fmt.Sprintf("https://reg_res/%s/value/%s", existingRes.Name, existingResValue1.Value)
 
-	got, err := s.db.PolicyClient.GetRegisteredResourceValue(s.ctx, &registeredresources.GetRegisteredResourceValueRequest{
-		Identifier: &registeredresources.GetRegisteredResourceValueRequest_Fqn{
-			Fqn: fqn,
+	testCases := []struct {
+		name string
+		req  *registeredresources.GetRegisteredResourceValueRequest
+	}{
+		{
+			name: "By ID",
+			req: &registeredresources.GetRegisteredResourceValueRequest{
+				Identifier: &registeredresources.GetRegisteredResourceValueRequest_Id{
+					Id: existingResValue1.ID,
+				},
+			},
 		},
-	})
-	s.Require().NoError(err)
-	s.NotNil(got)
-	s.Equal(existingResValue1.ID, got.GetId())
-	s.Equal(existingResValue1.Value, got.GetValue())
-	s.Equal(existingRes.ID, got.GetResource().GetId())
-	metadata := got.GetMetadata()
-	s.False(metadata.GetCreatedAt().AsTime().IsZero())
-	s.False(metadata.GetUpdatedAt().AsTime().IsZero())
-}
-
-func (s *RegisteredResourcesSuite) Test_GetRegisteredResourceValue_ByInvalidID_Fails() {
-	got, err := s.db.PolicyClient.GetRegisteredResourceValue(s.ctx, &registeredresources.GetRegisteredResourceValueRequest{
-		Identifier: &registeredresources.GetRegisteredResourceValueRequest_Id{
-			Id: invalidID,
+		{
+			name: "By FQN",
+			req: &registeredresources.GetRegisteredResourceValueRequest{
+				Identifier: &registeredresources.GetRegisteredResourceValueRequest_Fqn{
+					Fqn: fmt.Sprintf("https://reg_res/%s/value/%s", existingRes.Name, existingResValue1.Value),
```

Comment:
Would you mind please using the `lib/identifier` functionality here instead of Sprintf?

Reply:
`Sprintf` is just so much less verbose than this though...

```
fqn := identifier.FullyQualifiedRegisteredResourceValue{
	Name: existingRes.Name,
	Value: existingResValue1.Value,
}
fqnStr := x.FQN()
```

and it is a test...